wrapanapi/azure.py

```diff
@@ -639,16 +650,25 @@ def get_vm_vhd(self, vm_name, resource_group=None):
         return vhd_endpoint
 
     def get_network_interface(self, vm_name, resource_group=None):
+        """
+        Returns name of the Primary network interface attached to the Instance
+        """
         self.logger.info("Attempting to Retrieve Azure VM Network Interface {}".format(vm_name))
         azure_data = self.run_script(
-            "Get-AzureRmVm -ResourceGroup \"{}\" -Name \"{}\" | convertto-xml -as String"
-            .format(resource_group or self.resource_group, vm_name))
-        data = self.clean_azure_xml(azure_data)
-        status_value = json.loads(etree.parse(StringIO(data)).getroot().xpath(
-            "./Object/Property[@Name='NetworkProfileText']/text()")[0])
-        nic_uri = status_value['NetworkInterfaces'][0]['ReferenceUri']
-        self.logger.info("Returned URI was {}".format(nic_uri))
-        return os.path.split(urlparse.urlparse(nic_uri).path)[1]
+            """
+            $nic = Get-AzureRmVm -ResourceGroup \"{}\" -Name \"{}\"
+            $nic.NetworkProfile.NetworkInterfaces| convertto-json -Depth 2
+            """.format(resource_group or self.resource_group, vm_name))
+        data = json.loads(azure_data)
+        try:
+            vm_nic = data["Id"]
+        except TypeError:
+            for nic in data:
+                if nic["Primary"] is True:
+                    vm_nic = nic["Id"]
```

Comment:
vm_nic can be undefined after this line, therefore I would recommend probably setting it to None initially and handle the case of None after this exception block